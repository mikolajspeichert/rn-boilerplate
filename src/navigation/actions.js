export default {
  login: {
    MOVE: 'LOGIN_MOVE',
  },
  main: {
    MOVE: 'MAIN_MOVE',
  },
}
