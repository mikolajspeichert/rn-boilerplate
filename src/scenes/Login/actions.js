import NavigatorActions from '@navigation/actions'

const MOVE = 'MOVE'

export const actions = {
  MOVE,
}

export const moveToOnboarding = () => {
  return {
    type: NavigatorActions.login.MOVE,
    value: 'Onboarding',
  }
}
