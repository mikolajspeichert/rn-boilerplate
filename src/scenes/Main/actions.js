import NavigatorActions from '@navigation/actions'

const DEFAULT_ACTION = 'DEFAULT_ACTION'

export const actions = {
  DEFAULT_ACTION,
}

export const defaultAction = () => {
  return async dispatch => {
    dispatch({
      type: DEFAULT_ACTION,
      value: [],
    })
  }
}

export const moveToAdd = () => {
  return {
    type: NavigatorActions.MOVE,
    value: 'Add',
  }
}
export const moveToSettings = () => {
  return {
    type: NavigatorActions.MOVE,
    value: 'Settings',
  }
}
export const moveToLogin = () => {
  return {
    type: NavigatorActions.MOVE,
    value: 'Login',
  }
}
