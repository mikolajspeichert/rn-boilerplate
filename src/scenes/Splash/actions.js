import NavigatorActions from '@navigation/actions'

const DEFAULT_ACTION = 'DEFAULT_ACTION'

export const actions = {
  DEFAULT_ACTION,
}

export const moveToMain = () => {
  return async dispatch => {
    setTimeout(() => {
      dispatch({
        type: NavigatorActions.MOVE,
        value: 'Main',
      })
    }, 1000)
  }
}

export const moveToLogin = () => {
  return async dispatch => {
    setTimeout(() => {
      dispatch({
        type: NavigatorActions.MOVE,
        value: 'Login',
      })
    }, 1000)
  }
}
