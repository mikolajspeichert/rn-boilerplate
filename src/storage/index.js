import Realm from 'realm'
import Item from './item'

export default Realm.open({ schema: [Item] })
