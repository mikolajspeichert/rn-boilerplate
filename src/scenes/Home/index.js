import React, { Component } from 'react'
import { View, Text, Image, Platform, Button } from 'react-native'
import { connect } from 'react-redux'
import { createDefaultNotification } from '@services/notification'
import icons from '@assets/icons'
import styles from './styles'
import { loadItems } from './actions'
import I18n from '@i18n'

class HomeScreen extends Component {
  static navigationOptions = {
    title: I18n.t('home.hello'),
    tabBarIcon: <Image source={icons.check} style={styles.icon} />,
  }

  componentDidMount() {
    this.props.load()
  }

  render() {
    return (
      <View style={styles.home}>
        <Text>
          {I18n.t('home.welcome')} {Platform.OS === 'ios' ? 'iOS' : 'Android'}
        </Text>
        <Button onPress={() => createDefaultNotification()} title="Notify" />
      </View>
    )
  }
}
const mapStateToProps = state => {
  return {
    items: !!state.items && Object.values(state.items),
  }
}

const mapDispatchToProps = dispatch => ({
  goToMore: () => dispatch({ type: 'Add' }),
  load: () => dispatch(loadItems()),
})

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen)
