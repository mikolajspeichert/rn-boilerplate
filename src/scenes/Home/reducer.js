import { actions } from './actions'

var initialState = {
  items: [],
  theme: '#fafafa',
}

export default (state = initialState, action) => {
  switch (action.type) {
    case actions.LOAD_ITEMS:
      return Object.assign({}, state, {
        items: action.value,
      })
    default:
      return state
  }
}
