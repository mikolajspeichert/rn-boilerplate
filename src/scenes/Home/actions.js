import storage from '@storage'

const LOAD_ITEMS = 'LOAD_ITEMS'

export const actions = {
  LOAD_ITEMS,
}

export const loadItems = () => {
  return async dispatch => {
    // usually dispatch something that indicates fetching
    const realm = await storage
    dispatch({
      type: 'LOAD_ITEMS',
      value: realm.objects('Item'),
    })
  }
}
