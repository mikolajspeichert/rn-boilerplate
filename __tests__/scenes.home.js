import 'react-native'
import React from 'react'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import sinon from 'sinon'
import HomeScreen from '@scenes/Home'
import { loadItems } from '@scenes/Home/actions'
import storage from '@storage'

const state = {}
const mockStore = configureStore()

describe('Home Screen', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<HomeScreen store={mockStore(state)} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('dispatches navigation action', () => {
    console.log()
  })
})

describe('Home action', () => {
  beforeEach(async () => {
    let realm = await storage
    realm.write(() => {
      realm.create('Item', {
        id: 'first',
        title: 'FIRST_ITEM',
        notes: 'No notes on this one',
        created: new Date(1),
      })
      realm.create('Item', {
        id: 'second',
        title: 'Second item',
        notes: 'Some notes',
        created: new Date(100),
      })
    })
  })

  afterEach(async () => {
    let realm = await storage
    realm.write(() => {
      realm.delete(realm.objects('Item'))
    })

    realm.close()
  })

  it('loads items correctly', done => {
    const fakeDispatch = action => {
      expect(action).toMatchSnapshot()
      done()
    }
    loadItems()(fakeDispatch)
  })
})
