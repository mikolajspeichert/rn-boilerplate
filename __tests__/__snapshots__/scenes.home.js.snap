// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    [missing "en.home.welcome" translation]
     
    iOS
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#c12323",
            },
          ]
        }
      >
        [missing "en.home.more" translation]
      </Text>
    </View>
  </View>
</View>
`;

exports[`Home action loads items correctly 1`] = `
Object {
  "type": "LOAD_ITEMS",
  "value": Results {
    "0": Item {
      "created": 1970-01-01T00:00:00.001Z,
      "id": "first",
      "notes": "No notes on this one",
      "title": "FIRST_ITEM",
    },
    "1": Item {
      "created": 1970-01-01T00:00:00.100Z,
      "id": "second",
      "notes": "Some notes",
      "title": "Second item",
    },
  },
}
`;
